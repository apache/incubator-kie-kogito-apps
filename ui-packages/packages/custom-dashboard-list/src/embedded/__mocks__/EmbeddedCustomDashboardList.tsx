import React from 'react';

const MockedEmbeddedCustomDashboardList = (): React.ReactElement => {
  return <></>;
};

export default MockedEmbeddedCustomDashboardList;
