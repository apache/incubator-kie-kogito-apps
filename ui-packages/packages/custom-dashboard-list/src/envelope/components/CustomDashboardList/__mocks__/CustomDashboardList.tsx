import React from 'react';

const MockedCustomDashboardList = (): React.ReactElement => {
  return <></>;
};

export default MockedCustomDashboardList;
