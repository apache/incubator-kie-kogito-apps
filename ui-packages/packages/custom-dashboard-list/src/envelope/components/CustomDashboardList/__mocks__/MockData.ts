import { CustomDashboardInfo } from '../../../../api';

export const dashboardList: CustomDashboardInfo[] = [
  {
    name: 'age.dash.yaml',
    path: '/usr/home',
    lastModified: new Date('2021-08-23T13:26:02.13Z')
  },
  {
    name: 'products.dash.yaml',
    path: '/usr/home',
    lastModified: new Date('2021-08-23T13:26:02.13Z')
  },
  {
    name: 'age1.dash.yaml',
    path: '/usr/home',
    lastModified: new Date('2021-08-23T13:26:02.13Z')
  },
  {
    name: 'products1.dash.yaml',
    path: '/usr/home',
    lastModified: new Date('2021-08-23T13:26:02.13Z')
  }
];
