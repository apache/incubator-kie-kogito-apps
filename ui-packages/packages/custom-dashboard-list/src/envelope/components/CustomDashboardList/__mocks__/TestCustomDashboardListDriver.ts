import {
  CustomDashboardFilter,
  CustomDashboardInfo,
  CustomDashboardListDriver
} from '../../../../api';

export default class TestCustomDashboardListDriver
  implements CustomDashboardListDriver
{
  private readonly customDashboardInfos: CustomDashboardInfo[];
  constructor(customDashboardInfos: CustomDashboardInfo[]) {
    this.customDashboardInfos = customDashboardInfos;
  }
  applyFilter(customDashboardFilter: CustomDashboardFilter): Promise<void> {
    return Promise.resolve();
  }

  getCustomDashboardFilter(): Promise<CustomDashboardFilter> {
    return Promise.resolve({ customDashboardNames: ['age.dash.yaml'] });
  }

  getCustomDashboardsQuery(): Promise<CustomDashboardInfo[]> {
    return Promise.resolve(this.customDashboardInfos);
  }

  openDashboard(customDashboardInfo: CustomDashboardInfo): Promise<void> {
    return Promise.resolve();
  }
}
