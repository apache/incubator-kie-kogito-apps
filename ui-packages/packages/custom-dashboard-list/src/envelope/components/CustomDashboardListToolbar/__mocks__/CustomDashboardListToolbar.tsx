import React from 'react';

const MockedCustomDashboardListToolbar = (): React.ReactElement => {
  return <></>;
};

export default MockedCustomDashboardListToolbar;
