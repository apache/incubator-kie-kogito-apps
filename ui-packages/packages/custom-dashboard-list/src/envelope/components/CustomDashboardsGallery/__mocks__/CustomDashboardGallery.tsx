import React from 'react';

const MockedCustomDashboardGallery = (): React.ReactElement => {
  return <></>;
};

export default MockedCustomDashboardGallery;
