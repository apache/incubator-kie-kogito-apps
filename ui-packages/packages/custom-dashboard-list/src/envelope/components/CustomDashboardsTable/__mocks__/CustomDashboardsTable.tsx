import React from 'react';

const MockedCustomDashboardTable = (): React.ReactElement => {
  return <></>;
};

export default MockedCustomDashboardTable;
