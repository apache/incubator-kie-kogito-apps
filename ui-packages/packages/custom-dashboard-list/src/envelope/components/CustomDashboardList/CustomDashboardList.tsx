/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useEffect, useState } from 'react';
import { OUIAProps, componentOuiaProps } from '@kogito-apps/ouia-tools';
import { CustomDashboardListDriver } from '../../../api/CustomDashboardListDriver';
import CustomDashboardListToolbar from '../CustomDashboardListToolbar/CustomDashboardListToolbar';
import {
  Divider,
  Split,
  SplitItem,
  ToggleGroup,
  ToggleGroupItem
} from '@patternfly/react-core';
import {
  CustomDashboardInfo,
  CustomDashboardFilter
} from '../../../api/CustomDashboardListEnvelopeApi';
import CustomDashboardsTable from '../CustomDashboardsTable/CustomDashboardsTable';
import CustomDashboardsGallery from '../CustomDashboardsGallery/CustomDashboardsGallery';
import { BarsIcon, ThIcon } from '@patternfly/react-icons';
import { ServerErrors } from '@kogito-apps/components-common';
export interface CustomDashboardListProps {
  isEnvelopeConnectedToChannel: boolean;
  driver: CustomDashboardListDriver;
}

const CustomDashboardList: React.FC<CustomDashboardListProps & OUIAProps> = ({
  isEnvelopeConnectedToChannel,
  driver,
  ouiaId,
  ouiaSafe
}) => {
  const [filterNames, setFilterNames] = useState<string[]>([]);
  const [dashboardData, setDashboardData] = useState<CustomDashboardInfo[]>([]);
  const [isSelected, setIsSelected] = useState<{
    tableView: boolean;
    cardsView: boolean;
  }>({
    tableView: true,
    cardsView: false
  });
  const [isLoading, setIsLoading] = useState<boolean>(true);
  const [error, setError] = useState<any>(null);

  useEffect(() => {
    if (!isEnvelopeConnectedToChannel) {
      setIsLoading(true);
      return;
    }
    init();
  }, [isEnvelopeConnectedToChannel]);

  const init = async (): Promise<void> => {
    try {
      const namesFilter = await driver.getCustomDashboardFilter();
      setFilterNames(namesFilter.customDashboardNames);
      setIsLoading(true);
      const response = await driver.getCustomDashboardsQuery();
      setDashboardData(response);
      setIsLoading(false);
    } catch (errorResponse) {
      setError(errorResponse);
    }
  };

  const applyFilter = (filters: CustomDashboardFilter): void => {
    driver.applyFilter(filters);
    init();
  };

  const handleItemClick = (isChosen, event): void => {
    const toggleButtonId = event.currentTarget.id;
    if (toggleButtonId === 'tableView') {
      setIsSelected({
        tableView: true,
        cardsView: false
      });
    } else if (toggleButtonId === 'cardsView') {
      setIsSelected({
        tableView: false,
        cardsView: true
      });
    }
  };
  if (error) {
    return <ServerErrors error={error.message} variant={'large'} />;
  }

  return (
    <div {...componentOuiaProps(ouiaId, 'custom-dashboard-list', ouiaSafe)}>
      <Split hasGutter>
        <SplitItem>
          <CustomDashboardListToolbar
            applyFilter={applyFilter}
            setFilterDashboardNames={setFilterNames}
            filterDashboardNames={filterNames}
          />
        </SplitItem>
        <SplitItem isFilled></SplitItem>
        <SplitItem style={{ padding: '20px' }}>
          <ToggleGroup aria-label="switch view toggle group">
            <ToggleGroupItem
              icon={<BarsIcon />}
              aria-label="table view icon button"
              buttonId="tableView"
              isSelected={isSelected.tableView}
              onChange={handleItemClick}
            />
            <ToggleGroupItem
              icon={<ThIcon />}
              aria-label="card view icon button"
              buttonId="cardsView"
              isSelected={isSelected.cardsView}
              onChange={handleItemClick}
            />
          </ToggleGroup>
        </SplitItem>
      </Split>
      <Divider />
      {isSelected.tableView ? (
        <CustomDashboardsTable
          driver={driver}
          customDashboardData={dashboardData}
          setDashboardsData={setDashboardData}
          isLoading={isLoading}
        />
      ) : (
        <CustomDashboardsGallery
          customDashboardsDatas={dashboardData}
          driver={driver}
          isLoading={isLoading}
        />
      )}
    </div>
  );
};

export default CustomDashboardList;
