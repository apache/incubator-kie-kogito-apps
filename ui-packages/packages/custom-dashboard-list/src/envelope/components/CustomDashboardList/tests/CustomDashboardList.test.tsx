/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import CustomDashboardList from '../CustomDashboardList';
import { MockedCustomDashboardListDriver } from '../../../tests/mocks/MockedCustomDashboardsListDriver';
import { act } from 'react-dom/test-utils';
import wait from 'waait';
import TestCustomDashboardListDriver from '../__mocks__/TestCustomDashboardListDriver';
import { dashboardList } from '../__mocks__/MockData';

describe('customDashboard list tests', () => {
  jest.mock('../../CustomDashboardsGallery/CustomDashboardsGallery');
  jest.mock('../../CustomDashboardListToolbar/CustomDashboardListToolbar');
  jest.mock('../../CustomDashboardsTable/CustomDashboardsTable');
  Date.now = jest.fn(() => 1487076708000);
  const driver = new MockedCustomDashboardListDriver();

  it('envelope not connected to channel', async () => {
    const props = {
      isEnvelopeConnectedToChannel: false,
      driver: null
    };

    await act(async () => {
      render(<CustomDashboardList {...props} />);
    });

    expect(screen.getByText('Loading Dashboard...')).toBeTruthy();
  });

  it('render customDashboard list - table', async () => {
    const props = {
      isEnvelopeConnectedToChannel: true,
      driver: driver
    };
    let container;
    await act(async () => {
      container = render(<CustomDashboardList {...props} />).container;
    });
    expect(container).toMatchSnapshot();
  });

  /* Re-enable card view after thumbnails are available */
  /*it('render CustomDashboard list - gallery', async () => {
    const props = {
      isEnvelopeConnectedToChannel: true,
      driver: driver
    };
    let wrapper;
    // switches to gallery view
    await act(async () => {
      wrapper = mount(<CustomDashboardList {...props} />);
      wrapper.find(ToggleGroupItem).at(1).find('button').simulate('click');
    });
    await wait(0);
    await act(async () => {
      wrapper = wrapper.update();
    });
    expect(wrapper).toMatchSnapshot();

    // switches to table view
    await act(async () => {
      wrapper.find(ToggleGroupItem).at(0).find('button').simulate('click');
    });
    await wait(0);
    await act(async () => {
      wrapper = wrapper.update();
    });
    expect(wrapper.find('CustomDashboardList').exists()).toBeTruthy();
  });*/
});

let applyFilterMock;
let getCustomDashboardsQueryMock;
let props;
const getCustomDashboardListDriver = (
  items: number
): TestCustomDashboardListDriver => {
  const driver = new TestCustomDashboardListDriver(
    dashboardList.slice(0, items)
  );
  applyFilterMock = jest.spyOn(driver, 'applyFilter');
  getCustomDashboardsQueryMock = jest.spyOn(driver, 'getCustomDashboardsQuery');
  props.driver = driver;
  return driver;
};

describe('customDashboard list action tests', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    props = {
      isEnvelopeConnectedToChannel: true,
      driver: null
    };
  });

  it('CustomDashboard list - error page', async () => {
    getCustomDashboardListDriver(2);

    getCustomDashboardsQueryMock.mockImplementation(() => {
      throw new Error('404 error');
    });
    let wrapper;
    await act(async () => {
      const { container } = render(<CustomDashboardList {...props} />);
      wait();
      wrapper = container;
    });
    const error = wrapper.querySelector('h1')?.textContent;
    expect(error).toBe('Error fetching data');
  });

  it('CustomDashboard list - applyFilter and handleItemClick', async () => {
    getCustomDashboardListDriver(3);

    const { container } = render(<CustomDashboardList {...props} />);

    const toolbar = container.getElementsByClassName('pf-m-filter-group');

    expect(toolbar.length).toBe(1);

    /* Re-enable card view after thumbnails are available */
    /*const views = wrapper.find('ToggleGroupItem');
    expect(views.length).toBe(2);

    await act(() => {
      views.get(0).props['onChange'];
    });

    expect(views.get(0).props['isSelected']).toEqual(true);
    expect(views.get(1).props['isSelected']).toEqual(false);*/
  });
});
