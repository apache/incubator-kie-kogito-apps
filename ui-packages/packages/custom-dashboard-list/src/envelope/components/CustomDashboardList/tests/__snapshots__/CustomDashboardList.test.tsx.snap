// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customDashboard list tests render customDashboard list - table 1`] = `
<div>
  <div
    data-ouia-component-type="custom-dashboard-list"
    data-ouia-safe="true"
  >
    <div
      class="pf-l-split pf-m-gutter"
    >
      <div
        class="pf-l-split__item"
      >
        <div
          class="pf-c-toolbar pf-m-toggle-group-container"
          data-ouia-component-id="OUIA-Generated-Toolbar-2"
          data-ouia-component-type="custom-dashboard-list-toolbar"
          data-ouia-safe="true"
          id="custom-dashboard-list-with-filter"
        >
          <div
            class="pf-c-toolbar__content"
          >
            <div
              class="pf-c-toolbar__content-section"
            >
              <div
                class="pf-c-toolbar__group pf-m-toggle-group pf-m-show-on-xl"
              >
                <div
                  class="pf-c-toolbar__toggle"
                >
                  <button
                    aria-controls="custom-dashboard-list-with-filter-expandable-content-7"
                    aria-disabled="false"
                    aria-haspopup="false"
                    aria-label="Show Filters"
                    class="pf-c-button pf-m-plain"
                    data-ouia-component-id="OUIA-Generated-Button-plain-3"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="pf-c-toolbar__group pf-m-filter-group"
                >
                  <div
                    class="pf-c-toolbar__item"
                  >
                    <div
                      class="pf-c-input-group"
                    >
                      <input
                        aria-invalid="false"
                        aria-label="Dashboard name"
                        class="pf-c-form-control"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-4"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe="true"
                        data-testid="search-input"
                        id="customDashboardName"
                        name="customDashboardName"
                        placeholder="Filter by dashboard name"
                        type="search"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="pf-c-toolbar__item"
                  >
                    <button
                      aria-disabled="false"
                      class="pf-c-button pf-m-primary"
                      data-ouia-component-id="OUIA-Generated-Button-primary-2"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      data-testid="apply-filter"
                      id="apply-filter"
                      type="button"
                    >
                      Apply Filter
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="pf-c-toolbar__group pf-m-icon-button-group"
              >
                <div
                  class="pf-c-toolbar__item"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-plain"
                    data-ouia-component-id="OUIA-Generated-Button-plain-4"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="refresh"
                    id="refresh"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M440.65 12.57l4 82.77A247.16 247.16 0 0 0 255.83 8C134.73 8 33.91 94.92 12.29 209.82A12 12 0 0 0 24.09 224h49.05a12 12 0 0 0 11.67-9.26 175.91 175.91 0 0 1 317-56.94l-101.46-4.86a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12H500a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12h-47.37a12 12 0 0 0-11.98 12.57zM255.83 432a175.61 175.61 0 0 1-146-77.8l101.8 4.87a12 12 0 0 0 12.57-12v-47.4a12 12 0 0 0-12-12H12a12 12 0 0 0-12 12V500a12 12 0 0 0 12 12h47.35a12 12 0 0 0 12-12.6l-4.15-82.57A247.17 247.17 0 0 0 255.83 504c121.11 0 221.93-86.92 243.55-201.82a12 12 0 0 0-11.8-14.18h-49.05a12 12 0 0 0-11.67 9.26A175.86 175.86 0 0 1 255.83 432z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="pf-c-toolbar__expandable-content"
              id="custom-dashboard-list-with-filter-expandable-content-7"
            >
              <div
                class="pf-c-toolbar__group"
              />
            </div>
          </div>
          <div
            class="pf-c-toolbar__content pf-m-hidden"
            hidden=""
          >
            <div
              aria-hidden="true"
              class="pf-c-toolbar__group pf-m-hidden"
              hidden=""
            />
          </div>
        </div>
      </div>
      <div
        class="pf-l-split__item pf-m-fill"
      />
    </div>
    <hr
      class="pf-c-divider"
    />
    <div
      data-ouia-component-type="customDashboard-table"
      data-ouia-safe="true"
    >
      <table
        aria-label="Data Table"
        class="pf-c-table pf-m-grid-md"
        data-ouia-component-id="OUIA-Generated-Table-4"
        data-ouia-component-type="data-table"
        data-ouia-safe="true"
        role="grid"
      >
        <thead
          class=""
        >
          <tr
            class=""
            data-ouia-component-id="OUIA-Generated-TableRow-3"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <th
              aria-sort="none"
              class="pf-c-table__sort"
              data-key="0"
              data-label="Name"
              scope="col"
            >
              <button
                class="pf-c-table__button"
                type="button"
              >
                <div
                  class="pf-c-table__button-content"
                >
                  <span
                    class="pf-c-table__text"
                  >
                    Name
                  </span>
                  <span
                    class="pf-c-table__sort-indicator"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                      />
                    </svg>
                  </span>
                </div>
              </button>
            </th>
            <th
              aria-sort="descending"
              class="pf-c-table__sort pf-m-selected"
              data-key="1"
              data-label="Last Modified"
              scope="col"
            >
              <button
                class="pf-c-table__button"
                type="button"
              >
                <div
                  class="pf-c-table__button-content"
                >
                  <span
                    class="pf-c-table__text"
                  >
                    Last Modified
                  </span>
                  <span
                    class="pf-c-table__sort-indicator"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M168 345.941V44c0-6.627-5.373-12-12-12h-56c-6.627 0-12 5.373-12 12v301.941H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.569 9.373 33.941 0l86.059-86.059c15.119-15.119 4.411-40.971-16.971-40.971H168z"
                      />
                    </svg>
                  </span>
                </div>
              </button>
            </th>
          </tr>
        </thead>
        <tbody
          class=""
          role="rowgroup"
        >
          <tr
            class=""
            data-ouia-component-id="OUIA-Generated-TableRow-7"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <td
              class=""
              data-key="0"
              data-label="Name"
            >
              <a>
                <strong>
                  dashboard1
                </strong>
              </a>
            </td>
            <td
              class=""
              data-key="1"
              data-label="Last Modified"
            >
              <time
                datetime="NaN"
              >
                Invalid date
              </time>
            </td>
          </tr>
          <tr
            class=""
            data-ouia-component-id="OUIA-Generated-TableRow-8"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <td
              class=""
              data-key="0"
              data-label="Name"
            >
              <a>
                <strong>
                  dashboard2
                </strong>
              </a>
            </td>
            <td
              class=""
              data-key="1"
              data-label="Last Modified"
            >
              <time
                datetime="NaN"
              >
                Invalid date
              </time>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
